### ISSUE TITLE
Walk up parent directories to find the current crate

### ISSUE THREAD

**Issue by fitzgen** [rustwasm/wasm-pack#620]
When there isn't any crate directory specified, we should walk up directories until we see a Cargo.toml instead of giving up if the current directory has none.

I.e. we should make the first `wasm-pack test` attempt below Just Work:

```
~/src/dodrio/examples $ wasm-pack test --firefox --headless
Error: crate directory is missing a `Cargo.toml` file; is `.` the wrong directory?

~/src/dodrio/examples $ cd ..

~/src/dodrio $ wasm-pack test --firefox --headless
[INFO]: 🎯  Checking for the Wasm target...
   Compiling futures v0.1.26
   Compiling cfg-if v0.1.7
...
```

**Comment by ashleygwilliams** [comment]
i agree and i wish cargo fmt did this every day <3

**Comment by ashleygwilliams** [comment]
closed by #624
